account_bank_statement_import_adyen/tests/test_import_adyen.py

```diff
@@ -0,0 +1,119 @@
+# © 2017 Opener BV (<https://opener.amsterdam>)
+# © 2020 Vanmoof BV (<https://www.vanmoof.com>)
+# © 2015 Therp BV (<http://therp.nl>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import base64
+from odoo import tests
+from odoo.tests.common import TransactionCase
+from odoo.modules.module import get_module_resource
+
+
+@tests.tagged('standard', 'at_install')
+class TestImportAdyen(TransactionCase):
+    def setUp(self):
+        super(TestImportAdyen, self).setUp()
+        self.journal = self.env['account.journal'].create({
+            'company_id': self.env.user.company_id.id,
+            'name': 'Adyen test',
+            'code': 'ADY',
+            'type': 'bank',
+        })
+        self.journal.default_debit_account_id.reconcile = True
+        self.journal.write({
+            'adyen_merchant_account': 'YOURCOMPANY_ACCOUNT',
+            'update_posted': True,
+            'currency_id': self.ref('base.USD'),
+        })
+
+    def test01_import_adyen(self):
+        self._test_statement_import(
+            'account_bank_statement_import_adyen', 'adyen_test.xlsx',
+            'YOURCOMPANY_ACCOUNT 2016/48', self.journal)
+        statement = self.env['account.bank.statement'].search(
+            [], order='create_date desc', limit=1)
+        self.assertEqual(len(statement.line_ids), 22)
+        self.assertTrue(
+            self.env.user.company_id.currency_id.is_zero(
+                sum(line.amount for line in statement.line_ids)))
+
+        account = self.env['account.account'].search([(
+            'internal_type', '=', 'receivable')], limit=1)
+        for line in statement.line_ids:
+            line.process_reconciliation(new_aml_dicts=[{
+                'debit': -line.amount if line.amount < 0 else 0,
+                'credit': line.amount if line.amount > 0 else 0,
+                'account_id': account.id}])
+
+        statement.button_confirm_bank()
+        self.assertEqual(statement.state, 'confirm')
+        lines = self.env['account.move.line'].search([
+            ('account_id', '=', self.journal.default_debit_account_id.id),
+            ('statement_id', '=', statement.id)])
+        reconcile = lines.mapped('full_reconcile_id')
+        self.assertEqual(len(reconcile), 1)
+        self.assertTrue(lines.mapped('matched_debit_ids'))
+        self.assertTrue(lines.mapped('matched_credit_ids'))
+        self.assertEqual(lines, reconcile.reconciled_line_ids)
+
+        statement.button_draft()
+        self.assertEqual(statement.state, 'open')
+        self.assertFalse(lines.mapped('matched_debit_ids'))
```

Comment:
reconcile() call works and we get partial an full lines now. But, this is still failing. Do we want to check after button_draft() for the removal of partial reconciled lines? Or partial lines should stay?